### ISSUE TITLE
[Bug] RangeSlider (XCT compat renderer) is not shown using Grid

### ISSUE THREAD

**Issue by yunusefendi52** [dotnet/maui#2696]
### Description
I'm trying to use RangeSlider from xct using compat renderer with StackLayout as the parent and Grid for the whole layout but it's not shown in the layout.

### Steps to Reproduce

1. Create new maui app
2. Create grid with column definitions `Auto,Auto,Auto,Auto,*,auto,auto`
3. Create layout for the respective column definitions except the last one
4. Create layout in xaml inside Grid
```xaml
<StackLayout
            x:Name="Footer"
            Grid.Row="5">
            <ContentView
                HeightRequest="50"
                BackgroundColor="Red"/>

            <ContentView
                x:Name="ContentView"/>
            </StackLayout>
```
5. and then in code behind set ContentView to use RangeSlider
```cs
var range = new RangeSlider();
ContentView.Content = range;
```

### Expected Behavior
Layout is correctly resized and the range slider (compat renderer) is shown

### Actual Behavior
Layout is not correclty resized and the range slider (compat renderer) is not shown

### Basic Information

- Version with issue: dotnet 6.0.100-rc.1.21463.6
- Last known good version:
- IDE: vscode (with command line)
- Platform Target Frameworks: 
  - iOS:
  - Android: 12
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots

![Screen Shot 2021-09-25 at 6 47 58 AM](https://user-images.githubusercontent.com/19399214/134750286-d87e45f2-133e-443f-82d9-b01d0b685fef.png)

### Reproduction Link

[TestMaui.zip](https://github.com/dotnet/maui/files/7228841/TestMaui.zip)

**Comment by shimingsg** [comment]
@yunusefendi52 , I tried to repro with VS 17.2.0 Preview 2.0 [32303.288.main], but there is an exception blocked me to move forward
![image](https://user-images.githubusercontent.com/20982341/156683473-d2eca0d4-7ce1-4da2-9e9a-af879f6caddf.png)

**Comment by pictos** [comment]
@yunusefendi52 if you try using p13 and the latest XCT MauiCompat does it work?

**Comment by yunusefendi52** [comment]
@pictos let me try again this weekend

**Comment by yunusefendi52** [comment]
Tried again with p13 and the error is now "Handler not found for view RangeSlider", it was not error on previous version (as shown above), also tried with `AddCompatibilityRenderers` but still has the same issue

**Comment by pictos** [comment]
@yunusefendi52, thanks.
@VladislavAntonyuk, did you see any issue related to this on our repo?

**Comment by VladislavAntonyuk** [comment]
Yes, we have the same issue for expander https://github.com/xamarin/XamarinCommunityToolkit/issues/1788

**Comment by jfversluis** [comment]
Might be that this is with the `RangeSlider` in the toolkit and not so much .NET MAUI. In either case, there has gone some work in the layout for .NET MAUI, please check again if this is still important to you. I think for now this issue got stale and will close it for the time being.

Thanks!
